udata/core/spatial/models.py

```diff
@@ -100,15 +111,55 @@ def postal_string(self):
     def town_repr(self):
         """Representation of a town with optional county."""
         if self.county:
-            return '{name} <small>({county_name})</small>'.format(
-                name=self.name, county_name=self.county.name)
+            return ('{name} '
+                    '<small>(<a href="{county_url}">{county_name}</a>)</small>'
+                    '').format(name=self.name,
+                               county_url=self.county.url,
+                               county_name=self.county.name)
         return self.name
 
     @cached_property
-    def county(self):
+    def county_repr(self):
+        """Representation of a county."""
+        return '{name} <small>({code})</small>'.format(
+            name=self.name, code=self.code)
+
+    @cached_property
+    def region_repr(self):
+        """Representation of a region."""
+        return self.name
+
+    def get_parent(self, level):
         for parent in self.parents:
-            if parent.startswith('fr/county'):
-                return GeoZone.objects.get(id=parent)
+            if parent.startswith(level):
+                return GeoZone.objects.get(id=parent, level=level)
+
+    @cached_property
+    def county(self):
+        return self.get_parent('fr/county')
+
+    @cached_property
+    def region(self):
+        return self.get_parent('fr/region')
+
+    def get_children(self, level):
+        return GeoZone.objects(level=level, parents__in=[self.id])
+
+    @cached_property
+    def towns(self):
+        return self.get_children('fr/town').order_by('-population', '-area')
+
+    @cached_property
+    def counties(self):
+        return self.get_children('fr/county').order_by('-population', '-area')
+
+    @cached_property
+    def regions(self):
+        return self.get_children('fr/region').order_by('-population', '-area')
```

Comment:
Where do I know that children of a county should be towns? See my previous comment ;)